odoo/models.py

```diff
@@ -2904,6 +2875,8 @@ def _add_sql_constraints(self):
     def _execute_sql(self):
         """ Execute the SQL code from the _sql attribute (if any)."""
         if hasattr(self, "_sql"):
+            warnings.warn(f'_sql attribute is depreciated ({self._name}), override _auto_init instead',
+                          DeprecationWarning, stacklevel=2)
```

Comment:
Could be moved to the same place as the `_execute_sql` warning, to immediately warn anyone using `_sql`.